### ISSUE TITLE
`garden` importer is using `imp` which has been deprecated and slated for removal in `3.12`

### ISSUE THREAD

**Issue by hugovk** [kivy/kivy#8225]
**Describe the bug**

This project uses the [`imp` module](https://docs.python.org/3/library/imp.html) which has been deprecated since Python 3.4 and removed in 3.12:

* Raised `PendingDeprecationWarning` since [3.4 (2014)](https://github.com/python/cpython/commit/e4f41deccf94ccc798b1eb1f44657ade66669a60)
* Raised `DeprecationWarning` since [3.5 (2015)](https://github.com/python/cpython/commit/c0d91aff9a3b91307b26e8b7c34dfbf27bbdd43a)
* Updated `DeprecationWarning` to say removal in 3.12 since [3.10 (2021)](https://github.com/python/cpython/commit/dc6d3e1e4c0c1e4b2210edab8fb4762569dc2936)
* Removal planned for [3.12](https://github.com/python/cpython/issues/98040) [(2023)](https://github.com/python/cpython/pull/98573)

Python 3.12 is [set for release on 2023-10-02](https://devguide.python.org/versions/) and this library is one of the [top 5,000 most-downloaded from PyPI](https://hugovk.github.io/top-pypi-packages/).

Please could you upgrade to use `importlib`? The [`imp` docs](https://docs.python.org/3.11/library/imp.html) have suggestions on what to use to replace each function and constant.

**Expected behavior**

Compatible with importlib and Python 3.12.

**To Reproduce**

`git grep \bimp\b`

**Comment by misl6** [comment]
Hi @hugovk !

Can you please post the log snippet which is raising this warning?

**Comment by hugovk** [comment]
Hello! I found it by searching the code:

https://github.com/kivy/kivy/blob/b69ca46a09652893a9f022387865432883838bbc/kivy/garden/__init__.py#L183-L184

**Comment by misl6** [comment]
Ok, thank you! Changed the issue title.

**Comment by JoaoRoizenFontana** [comment]
Hi!
I'm new here. I would like to work on this issue.

**Comment by misl6** [comment]
Hi @JoaoRoizenFontana !

You've been assigned to this issue 😀

**Comment by JoaoRoizenFontana** [comment]
Thank you!! @misl6

**Comment by JoaoRoizenFontana** [comment]
Hi! @misl6 

I followed the instructions and studied the importlib docs in order to make the necessary changes:

1. Identified the usage of the deprecated imp module in the original code.
-The original code used imp.load_module to load modules dynamically, which is deprecated as of Python 3.4.

2. Researched the recommended alternative for loading modules dynamically.
-The recommended alternative is to use the importlib module, which provides functions for working with modules and importing them dynamically.

3. Identified the specific importlib functions to replace the deprecated imp functions.
-To replace imp.load_module, I used importlib.util.spec_from_file_location to generate a module specification and importlib.util.module_from_spec to create the module object.
